src/rad/appdata.cpp

```diff
@@ -1319,16 +1321,41 @@ bool ApplicationData::LoadProject( const wxString &file, bool justGenerate )
 
 		if ( older )
 		{
+			wxLogDebug("Old version: %d.%d -- current %d.%d", fbpVerMajor, fbpVerMajor, m_fbpVerMajor, m_fbpVerMinor);
+
 			if( justGenerate ){
 				wxLogError( wxT( "This project file is out of date.  Update your .fbp before using --generate" ) );
 				return false;
 			}
-			if ( wxYES == wxMessageBox( wxT( "This project file is not of the current version.\n" )
-			                            wxT( "Would you to attempt automatic conversion?\n\n" )
-			                            wxT( "NOTE: This will modify your project file on disk!" ), _( "Old Version" ), wxYES_NO ) )
+
+			wxFileName fname(file);
+			wxString   strMsg, strOldBackupName(fname.GetFullName() + wxT(".bak"));
+
+			if (wxMessageBox(strMsg.Format(_("This project file is an old version.\n\n"
+			                                 "Do you want to copy the old version to %s and save an updated version?"),
+			                               strOldBackupName),
+			                 _("Old project file"), wxYES_NO) == wxYES)
 			{
-				// we make a backup of the project
-				::wxCopyFile( file, file + wxT( ".bak" ) );
+				{
+					wxLogNull logNo;  // Prevents wxCopyFile from displaying a wxWidgets error.
+					fname = file + wxT(".bak");
+
+					while (!::wxCopyFile(file, fname.GetFullPath()))
+					{
+						wxRichMessageDialog dlg(
+						    NULL, strMsg.Format(_("Unable to save old version to %s.\n\n"), strOldBackupName),
```

Comment:
Don't create a dialog without a parent, especially not a modal one. Don't call that static method through an object. Actually, why do you create that message in every loop? It should be constant and simply assigned outside the loop to strMsg.